c_emulator/riscv_sail.h

```diff
@@ -68,3 +67,13 @@ struct zMcause {
 extern struct zMcause zmcause, zscause;
 
 extern mach_bits zminstret;
+
+enum kind_zphysaddr { Kind_zphysaddr };
```

Comment:
Having to duplicate this sail-internal representation does not sound like a good idea to me. IMO the C callback should just use the underlying bits.

Reply:
If/when we switch to the `struct Model { #include "model.c" }` trick then we won't have to do this. But I agree, it's too error-prone until it's actually checked by the compiler.